### PR TITLE
Update codegen based on recent azcore refactor

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Rename package prefixes based on the refactor.
Updated telemetry policy codegen to reflect the recent changes.
Changed telemetry const generation per policy changes.
Updated tests to use azcore/to.

### CODE REVIEW

File: test/autorest/azurespecialsgroup/zz_generated_constants.go [hunk 1]
```diff
@@ -9,6 +9,6 @@
 package azurespecialsgroup
 
 const (
-	telemetryInfo = "azsdk-go-azurespecialsgroup/" + version
-	version       = "v0.1"
+	module  = "azurespecialsgroup"
```
Comment (RickWinter) [inline test/autorest/azurespecialsgroup/zz_generated_constants.go:12]: Should this be azspecialsgroup for the package?

File: test/autorest/bytegroup/zz_generated_constants.go [hunk 2]
```diff
@@ -9,6 +9,6 @@
 package bytegroup
 
 const (
-	telemetryInfo = "azsdk-go-bytegroup/" + version
-	version       = "v0.1"
+	module  = "bytegroup"
```
Comment (RickWinter) [inline test/autorest/bytegroup/zz_generated_constants.go:12]: Should these have a package name that makes them unique to tests?  aztestbytegroup, etc...?

File: test/synapse/2019-06-01/azspark/zz_generated_constants.go [hunk 3]
```diff
@@ -9,8 +9,8 @@
 package azspark
 
 const (
-	telemetryInfo = "azsdk-go-azspark/" + version
-	version       = "v0.1"
+	module  = "azspark"
+	version = "v0.1"
```
Comment (RickWinter) [inline test/synapse/2019-06-01/azspark/zz_generated_constants.go:13]: We should standardize our versions v0.1.0 keeping the Major.Minor.Rev

**Review by RickWinter — APPROVED** [review]

File: test/autorest/bytegroup/zz_generated_constants.go (hunk 2 above)
Comment (jhendrixMSFT) [reply test/autorest/bytegroup/zz_generated_constants.go:12]: We kept the package naming consistent with track 1 so it was easier (at least in the beginning) to use track 1 as a baseline.  While we could go and change it, it's a bit of busy work that doesn't bring much value IMO.

File: test/synapse/2019-06-01/azspark/zz_generated_constants.go (hunk 3 above)
Comment (jhendrixMSFT) [reply test/synapse/2019-06-01/azspark/zz_generated_constants.go:13]: fixed

**Pull request merged**